### ISSUE TITLE
[bug] Proxy headers with uppercase letters are no longer passed to the proxy endpoint 

### ISSUE THREAD

**Issue by hmnhf** [centrifugal/centrifugo#572]
**Describe the bug**. A clear and concise description of what the bug is.

After upgrading to `v4.0.2` from `v4.0.0`, Centrifugo stopped passing the configured proxy headers to the proxy endpoint.
I noticed #558 in [v4.0.2 changelogs](https://github.com/centrifugal/centrifugo/blob/master/CHANGELOG.md#improvements) and changed the configured headers in `config.json` to have all of their letters lowercased. (E.g. from `Foo-Bar-Baz` to `foo-bar-baz`.)

After the change, Centrifugo started passing the proxy headers again. (Double-checked this by reverting and re-applying the config change to make sure the problem is with the uppercase letters.)

**Versions**

Centrifugo version is 4.0.2
Client library used is centrifuge-go of version  v0.9.3
Operating system is macOS

**Steps to Reproduce** How can the bug be triggered?

1. Add a connect proxy configuration similar to this:

```
{
  "connect_proxy_name": "connect"
  "proxies": [
    {
      "name": "connect",
      "endpoint": "https://cent-proxy.example.com/connect",
      "http_headers": ["Foo-Bar-Baz", "Qux-Fred-Thud"]
    }
 }
 ```

2.  Send a connect request to the Centrifugo server with `Foo-Bar-Baz` and `Qux-Fred-Thud` headers set.

**Expected behavior** What output or behaviour were you expecting instead?

Centrifugo should pass `Foo-Bar-Baz` and `Qux-Fred-Thud` headers to the proxy endpoint, but it doesn't.

**Comment by FZambia** [comment]
@hmnhf hello, thanks for a detailed bug report!

Very sorry, this should be fixed with #573  - the issue should only happen if `granular_proxy_mode` is used.

**Comment by FZambia** [comment]
Fixed in Centrifugo v4.0.3

**Comment by hmnhf** [comment]
@FZambia Awesome, thanks!
